### ISSUE TITLE
Wrong ridge_objective function in the ridge_reg_implicit_diff example

### ISSUE THREAD

**Issue by raphaelreinauer** [google/jaxopt#9]
There is an error in the ridge_objective function:
`return 0.5 * jnp.mean(residuals ** 2) + 0.5 * lam * jnp.sum(params ** 2)`
https://github.com/google/jaxopt/blob/2dda8c3dfcda82f9794c801ec50d1d4386a50114/examples/ridge_reg_implicit_diff.py#L33

As correctly stated in Figure 1 in the paper https://arxiv.org/pdf/2105.15183.pdf
the ridge objective should be
`return 0.5 * jnp.sum(residuals ** 2) + 0.5 * lam * jnp.sum(params ** 2)`
i.e. change the jnp.mean to jnp.sum.

**Comment by mblondel** [comment]
Using `jnp.mean` (which corresponds to dividing the loss term by the number of samples) is pretty standard as well. In JAXopt, we tend to use that convention because then the loss can be used on a mini batch and `jax.grad` gives a stochastic gradient.

**Comment by raphaelreinauer** [comment]
Thanks for the explanation! That why you have a `len(y_tr)` factor in
https://github.com/google/jaxopt/blob/2dda8c3dfcda82f9794c801ec50d1d4386a50114/examples/ridge_reg_implicit_diff.py#L46
